romancal/dq_init/dq_init_step.py

```diff
@@ -0,0 +1,99 @@
+#! /usr/bin/env python
+
+from romancal.stpipe import RomanStep
+from romancal.dq_init import dq_initialization
+from roman_datamodels.datamodels import RampModel
+import roman_datamodels as rdm
+from roman_datamodels.testing import utils as testutil
+
+__all__ = ["DQInitStep"]
+
+class DQInitStep(RomanStep):
+    """Initialize the Data Quality extension from the
+    mask reference file.
+
+    The dq_init step initializes the pixeldq attribute of the
+    input datamodel using the MASK reference file.  For some
+    Guiding and Image model types, initalize the dq attribute of
+    the input model instead.  The dq attribute of the MASK model
+    is bitwise OR'd with the pixeldq (or dq) attribute of the
+    input model.
+    """
+
+
+    reference_file_types = ['mask']
+
+    def process(self, input):
+        """Perform the dq_init calibration step
+
+        Parameters
+        ----------
+        input : Roman datamodel
+            input roman datamodel
+
+        Returns
+        -------
+        output_model : Roman datamodel
+            result roman datamodel
+        """
+        # Open datamodel
+        input_model = self.open_model(input)
+
+        # Convert to RampModel if needed
+        if not (isinstance(input_model, RampModel)):
+            # Create base ramp node with dummy values (for validation)
+            input_ramp = testutil.mk_ramp(input_model.shape)
```

Comment:
Is this section here and are you calling `testutil` only to deal with the fact that you need to turn a 3D array into a 4D array?

Reply:
No. I call it in order to create a base ramp node. Simply opening the input_model as a RampModel wouldn't work right, no matter how many ways I tried it.. properties deep within the model never "changed over" (like the taggedobjectnode, uri, etc. - something Perry clued me in on), and it never added any metadata keywords required by ramp_model but not by the input (which would then make it fail on validation). The code here was the tidiest way I could get the input into a proper RampModel.

Reply:
New issue added to RDM for this.